expression/util.go

```diff
@@ -526,3 +527,55 @@ func GetParamExpression(ctx sessionctx.Context, v *driver.ParamMarkerExpr, useCa
 	}
 	return value, nil
 }
+
+// ConvertToByItemExpr rewrites ByItem.ExprNode to a proper ExprNode.
+func ConvertToByItemExpr(n ast.Node) ast.Node {
+	switch v := n.(type) {
+	case *driver.ParamMarkerExpr:
+		return &ast.PositionExpr{P: v}
+	}
+	return n
+}
+
+// PosFromPositionExpr generates a position value from PositionExpr.
+func PosFromPositionExpr(ctx sessionctx.Context, v *ast.PositionExpr) (int, bool, error) {
+	if v.P == nil {
+		return v.N, false, nil
+	}
+	value, err := GetParamExpression(ctx, v.P.(*driver.ParamMarkerExpr))
+	if err != nil {
+		return 0, true, errors.Trace(err)
```

Comment:
Once it's traced, we don't need to trace it anymore.
So here we can just return err.

Reply:
Thank you for your detailed comments. The updated PR will address the issue.